### PR TITLE
Keypair and Mnemonic modules

### LINKED ISSUE #33: Add KeyPair boundary
### Describe the solution you'd like

Split the `Encryption` module and add KeyPair and Mnemonic related functions to these modules: `KeyPair`, `Mnemonic`.

### Describe alternatives you've considered

Bring the `random_keypair` and `public_key_from_secret_key` functions to `KeyPair` module and remove the `mnemonic_encryopt` from `Encryption`.

### PR DESCRIPTION
**Pull request by peterchicaiza**
## Description
Removes functions related with Keypairs and Mnemonic phrases from the Encryption module and locates them in the corresponding module.

ℹ️ This PR closes #33

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by miguelnietoa — APPROVED** [review]

File: apps/mintacoin/lib/mintacoin/keypair/default.ex [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    |> :crypto.generate_key(@edwards_curve_ed, secret_key)
+    |> encode_keypair()
+  rescue
+    _error -> {:error, :decoding_error}
+  end
+
+  @spec encode_keypair(keypair :: keypair()) :: {:ok, keypair()}
+  defp encode_keypair({public_key, secret_key}) do
+    encoded_public_key = Base.hex_encode32(public_key, padding: false)
+    encoded_secret_key = Base.hex_encode32(secret_key, padding: false)
```
Comment (juanhurtado10) [inline apps/mintacoin/lib/mintacoin/keypair/default.ex:37]: @peterchicaiza secret_key IMO, you should use a `Base.encode64` for the secret

File: apps/mintacoin/lib/mintacoin/keypair/spec.ex [hunk 2]
```diff
@@ -0,0 +1,14 @@
+defmodule Mintacoin.Keypair.Spec do
+  @moduledoc """
+  Specifies the functions available for the Keypair module
+  """
+
+  @type public_key() :: String.t()
+  @type secret_key() :: String.t()
+  @type keypair() :: {public_key(), secret_key()}
+  @type error() :: {:error, :decoding_error | :encryption_error}
```
Comment (juanhurtado10) [inline apps/mintacoin/lib/mintacoin/keypair/spec.ex:9]: @peterchicaiza what about something like ...

```elixir
 @type error() :: :decoding_error | :encryption_error
..
..
@callback from_secret_key(secret_key()) :: {:ok, keypair()} | {:error, error()}
```

**Review by juanhurtado10 — CHANGES_REQUESTED** [review]

File: apps/mintacoin/lib/mintacoin/keypair/default.ex (hunk 1 above)
Comment (peterchicaiza) [reply apps/mintacoin/lib/mintacoin/keypair/default.ex]: Using encode64 could be useful for encrypting 👌🏼

File: apps/mintacoin/lib/mintacoin/keypair/spec.ex (hunk 2 above)
Comment (peterchicaiza) [reply apps/mintacoin/lib/mintacoin/keypair/spec.ex]: ok, it might be more readable, done 👍🏼

**Pull request merged**